File: week2-math/problem05-약수의합/heetae.java
/**
문제: 약수의 합
접근 방식: 어짜피 루트만큼 탐색하면 반대편을 알 수 있으므로 i*i만큼만 돌면 된
시간복잡도: O(√n)
*/
class Solution {
    public int solution(int n) {
        int answer = 0;

        for (int i = 1; i * i <= n; i++) {
            if (n%i  == 0) {
                answer += i;

                if (i != n/i) {
                    answer += n/i;
                }
            }
        }

        return answer;
    }
}

File: week1-array-string/problem11-문자열나누기/heetae.py
# 문제: 문자열 나누기
# 접근 방식: 각 부분 문자열의 첫 문자와 같은 문자와 다른 문자의 개수를 세고, 두 개수가 같아질 때마다 분리한다.
# 시간복잡도: O(n) (n은 s의 길이)

def solution(s):
    answer = 0
    first = 0
    second = 0
    f_index = -1

    for i in range(0, len(s)):
        if f_index == -1:
            f_index = i
            first = 1
            continue
        else:
            if s[i] == s[f_index]:
                first += 1
            else:
                second += 1
        if first == second:
            answer += 1
            f_index = -1
            first = 0
            second = 0
    if f_index != -1:
        answer += 1
    return answer
